docs/theory/swe.md

```diff
@@ -0,0 +1,342 @@
+over the domain $\Omega$ and applying the (two-dimensional) divergence theorem
+to the flux term.
+
+To compare these two integral forms, consider the line integral in $(\ref{3})$
+on a quadrilateral cell along a counterclockwise path connecting its vertices:
+
+\begin{eqnarray}
+I_1 =  &\oint_{\partial\Omega_i}& \left( \mathbf{E}~dy - \mathbf{G}~dx \right) \\
+   = - &\int_{x_1}^{x_2}&\mathbf{G}|_{y_1}~dx ~~~~\mathrm{(bottom)} \\
+     + &\int_{y_1}^{y_2}&\mathbf{E}|_{x_2}~dy ~~~~\mathrm{(right)} \\
+     - &\int_{x_2}^{x_1}&\mathbf{G}|_{y_2}~dx ~~~~\mathrm{(top)} \\
+     + &\int_{y_2}^{y_1}&\mathbf{E}|_{x_1}~dy ~~~~\mathrm{(left)}
+\end{eqnarray}
+
+Compare this to the corresponding term in $(\ref{4})$, a surface integral in
+which the areas (lengths) of the faces (edges) are always positive:
+
+\begin{eqnarray}
+I_2 = &\iint_{\partial\Omega_i}& \mathbf{\vec{F}} \cdot \vec{n}~dl_i \\
+     - &\int_{x_1}^{x_2}&\mathbf{F}_y|_{y_1}~dx ~~~~\mathrm{(bottom)} \\
+     + &\int_{y_1}^{y_2}&\mathbf{F}_x|_{x_2}~dy ~~~~\mathrm{(right)} \\
+     + &\int_{x_1}^{x_2}&\mathbf{F}_y|_{y_2}~dx ~~~~\mathrm{(top)} \\
+     - &\int_{y_1}^{y_2}&\mathbf{F}_x|_{x_1}~dy ~~~~\mathrm{(left)}
+\end{eqnarray}
+
+Evidently $I_1 = I_2$ if $\mathbf{F}_x = \mathbf{E}$ and $\mathbf{F}_y = \mathbf{G}.$
+
+In the rest of this section, we use the flux form $(\ref{4})$ of the shallow
+water equations.
+
+### Finite volume formulation
+
+We can obtain a finite volume method for these equations by defining
+_horizontally-averaged_ quantities for flow depth and velocities:
+
+\begin{eqnarray}
+h_i &=& \frac{1}{A_i}\int_{\Omega_i} h d\Omega_i \\
+u_i &=& \frac{1}{A_i}\int_{\Omega_i} u d\Omega_i \\
+v_i &=& \frac{1}{A_i}\int_{\Omega_i} v d\Omega_i
+\end{eqnarray}
+
+where $A_i = \int_{\Omega_i}d\Omega_i$ is the area enclosed within cell $i$. We
+also introduce the _horizontally-averaged solution vector_
+
+$$
+\mathbf{U}_i = (h_i, h_i u_i, h_i v_i)^T
+$$
+
+and the _horizontally-averaged source vector_
+
+$$
+\mathbf{S}_i = \frac{1}{A_i}\int_{\Omega_i} \mathbf{S} d\Omega_i.
+$$
+
+Finally, we define the _face-averaged normal flux vector_ between cell $i$ and
+an adjoining cell $j$:
+
+$$
+\mathbf{F}_{ij} = \frac{1}{l_{ij}}\int_{\partial\Omega_i\bigcap\partial\Omega_j}\mathbf{\vec{F}}\cdot\vec{n}~dl \label{5}
+$$
+
+where $l_{ij}$ is the length of the face connecting cells $i$ and $j$.
+
+With these definitions, the shallow water equations in cell $i$ are
+
+$$
+\frac{\partial\mathbf{U}_i}{\partial t} + \sum_j\mathbf{F}_{ij} l_{ij} = \mathbf{S}_i, \label{6}
+$$
+
+where the index $j$ in each term of the sum refers to a neighboring cell of cell $i$.
+
+### Boundary conditions
+
+To incorporate boundary conditions, we partition the domain boundary $\Gamma$
+into disjoint line segments, each of which represents a _boundary face_
+$\Gamma_i$. Every cell $j$ that touches the boundary $\Gamma$ has at least one
```

Comment:
How about we introduce $\Gamma$ as a face and say the domain consists of the following?
- Internal faces: $\Gamma^{int}$, and
- Boundary faces: $\Gamma^{bnd}$

Reply:
I've been using $\Gamma$ to refer to the computational domain boundary, so I personally think it's more specific to use $\Gamma_f$ (for example) to refer to boundary face $f$. The interior faces all belong to the boundary of some cell $\partial\Omega_i$, following the convention that $\partial$ before a symbol denotes the boundary of that symbol. Maybe we can find a more succinct way to express it, though. I'll think about it.

Reply:
Okay, I understand what you are saying. My intent here was to see if we can introduce a symbol that simplifies the following notation:

- ${\partial\Omega_j\subset\Gamma}$ 
- ${\partial\Omega_j\not\subset\Gamma}$

Reply:
Maybe I can annotate the terms in the equation with `(boundary faces)` and `(interior faces)` to make it more clear.